File: app/src/main/java/com/theone/music/ui/fragment/rank/RankItemFragment.java
package com.theone.music.ui.fragment.rank;//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import android.os.Bundle;

import com.theone.common.constant.BundleConstant;
import com.theone.music.ui.fragment.BaseMusicFragment;
import com.theone.music.viewmodel.RankViewModel;

/**
 * @author The one
 * @date 2022-04-11 14:31
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
public class RankItemFragment extends BaseMusicFragment<RankViewModel> {

    private RankItemFragment(){}

    public static RankItemFragment newInstance(int type){
        RankItemFragment fragment = new RankItemFragment();
        Bundle bundle = new Bundle();
        bundle.putInt(BundleConstant.TYPE,type);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void initData() {
        int type = getArguments().getInt(BundleConstant.TYPE);
        getViewModel().setType(type);
    }
}

File: app/src/main/java/com/theone/music/ui/fragment/rank/RankItemFragment.kt
package com.theone.music.ui.fragment.rank

import com.theone.common.constant.BundleConstant
import com.theone.common.ext.bundle
import com.theone.common.ext.getValueNonNull
import com.theone.music.net.NetConstant
import com.theone.music.ui.fragment.BaseMusicFragment
import com.theone.music.ui.fragment.MusicItemFragment
import com.theone.music.viewmodel.MusicViewModel
import com.theone.music.viewmodel.RankViewModel

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛
/**
 * @author The one
 * @date 2022-04-07 08:50
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
class RankItemFragment private constructor() :
    BaseMusicFragment<RankViewModel>() {

    companion object {
        fun newInstance(type: Int): RankItemFragment = RankItemFragment().bundle {
            putInt(BundleConstant.TYPE, type)
        }
    }

    private val mType: Int by getValueNonNull(BundleConstant.TYPE)

    override fun initData() {
        getViewModel().type = mType
    }
}